engine/access/rpc/backend/backend_transactions.go

```diff
@@ -273,16 +277,19 @@ func (b *backendTransactions) GetTransactionResult(
 
 	// access node may not have the block if it hasn't yet been finalized, hence block can be nil at this point
 	if block != nil {
-		blockID = block.ID()
-		transactionWasExecuted, events, statusCode, txError, err = b.lookupTransactionResult(ctx, txID, blockID)
-		blockHeight = block.Header.Height
+		transactionWasExecuted, events, statusCode, txError, err = b.lookupTransactionResult(ctx, txID, block.ID())
 		if err != nil {
 			return nil, rpc.ConvertError(err, "failed to retrieve result from any execution node", codes.Internal)
 		}
 
 		// an additional check to ensure the correctness of the collection ID.
 		expectedCollectionID, err := b.lookupCollectionIDInBlock(block, txID)
-		if err != nil {
+
+		// if the collection has not been indexed yet, the lookup will return a not found error.
+		// if the request included a blockID or collectionID in its the search criteria, not found
+		// should result in an error because it's not possible to guarantee that the result found
+		// is the correct one.
+		if err != nil && (blockID != flow.ZeroID || collectionID != flow.ZeroID) {
```

Comment:
can we simply this condition? or break it into multiple if checks? it's too complex.

Reply:
what would you want it to be? 
```golang
if err != nil {
	if blockID != flow.ZeroID || collectionID != flow.ZeroID {
		return nil, rpc.ConvertStorageError(err)
	}
}
```